1.5/Defs/Buildings/Buildings_Structure.xml

```diff
@@ -340,4 +340,147 @@
+            <Steel>20</Steel>
+            <WoodLog>20</WoodLog>
+        </costList>
+        <uiIconPath>Things/Building/PS_BugWallStorage_UI</uiIconPath>
+        <uiIconScale>0.85</uiIconScale>
+        <modExtensions>
+            <li Class="PawnStorages.PSExtension" />
+        </modExtensions>
+    </ThingDef>
+
+    <ThingDef ParentName="BuildingBase">
+        <defName>PS_BugWallCocoon</defName>
+        <label>Cocoon</label>
+        <description>An insect like cocoon that can bind and suspend a pawn. It provides nutrients while keeping the pawn in a semi hibernated state.\n\nThe pawns hediffs will keep ticking, but age and hunger do not.</description>
+        <thingClass>PawnStorages.Building_PawnStorage</thingClass>
+        <rotatable>true</rotatable>
+        <graphicData>
+            <texPath>Things/Building/PS_BugCocoon</texPath>
+            <graphicClass>Graphic_Single</graphicClass>
+            <drawSize>(1,2)</drawSize>
+        </graphicData>
+        <size>(1,2)</size>
+        <tickerType>Normal</tickerType>
+        <altitudeLayer>Building</altitudeLayer>
+        <passability>PassThroughOnly</passability>
+        <researchPrerequisites IsNull="True" />
+        <pathCost>100</pathCost>
+        <overrideMinifiedRot>North</overrideMinifiedRot>
+        <minifiedDrawOffset>(0, 0, 0.2)</minifiedDrawOffset>
+        <uiOrder>2991</uiOrder>
+        <designationCategory>Furniture</designationCategory>
+        <containedPawnsSelectable>true</containedPawnsSelectable>
+        <comps>
+            <li Class="PawnStorages.CompProperties_PSAssignableToPawn">
+                <drawAssignmentOverlay>false</drawAssignmentOverlay>
+                <maxAssignedPawnsCount>1</maxAssignedPawnsCount>
+                <colonyAnimalsOnly>true</colonyAnimalsOnly>
+                <showGizmo>true</showGizmo>
+                <pawnDrawScaleX>1</pawnDrawScaleX>
+                <pawnDrawScaleZ>0.5</pawnDrawScaleZ>
+                <drawGrayscale>false</drawGrayscale>
+                <cameraZoom>1.5</cameraZoom>
+            </li>
+            <li Class="PawnStorages.TickedStorage.CompProperties_TickedStorage">
+                <TickAge>false</TickAge>
+                <TickHediffs>true</TickHediffs>
+                <needsDrop>false</needsDrop>
+                <releaseOption>true</releaseOption>
+                <releaseAllOption>true</releaseAllOption>
+                <maxStoredPawns>1</maxStoredPawns>
+                <convertOption>true</convertOption>
+                <canBeScheduled>true</canBeScheduled>
+                <showStoredPawnHeadOnly>true</showStoredPawnHeadOnly>
+                <showStoredPawn>true</showStoredPawn>
+            </li>
+        </comps>
+        <statBases>
+            <WorkToBuild>600</WorkToBuild>
+            <MaxHitPoints>50</MaxHitPoints>
+            <Flammability>0.5</Flammability>
+            <Mass>5</Mass>
+        </statBases>
+        <costList>
+            <Steel>20</Steel>
```

Comment:
I don't think these should need steel, probably not wood either bc caves don't really have it. I'm thinking just some insect jelly maybe?

Reply:
Yup, placeholder. Insect jelly makes sense.